src/multilayerqg.jl

```diff
@@ -211,6 +214,49 @@ struct SingleLayerParams{T, Aphys3D, Aphys2D, Trfft} <: BarotropicParams
   rfftplan :: Trfft
 end
 
+"""
+    TwoLayerParams{T, Aphys3D, Aphys2D, Trfft}(g, f₀, β, ρ, H, U, eta, μ, ν, nν, calcFq!, g′, Qx, Qy, rfftplan)
+
+A struct containing the parameters for the TwoLayerQG problem. Included are:
+
+$(TYPEDFIELDS)
+"""
+struct TwoLayerParams{T, Aphys3D, Aphys2D, Trfft} <: AbstractParams
+  # prescribed params
+    "gravitational constant"
+         g :: T
+    "constant planetary vorticity"
+        f₀ :: T
+    "planetary vorticity y-gradient"
+         β :: T
+    "array with density of each fluid layer"
+         ρ :: Aphys3D
+    "tuple with rest height of each fluid layer"
+         H :: Tuple
+   "array with imposed constant zonal flow U(y) in each fluid layer"
+         U :: Aphys3D
+    "array containing topographic PV"
+       eta :: Aphys2D
+    "linear bottom drag coefficient"
+         μ :: T
+    "small-scale (hyper)-viscosity coefficient"
+         ν :: T
+    "(hyper)-viscosity order, `nν```≥ 1``"
+        nν :: Int
+    "function that calculates the Fourier transform of the forcing, ``F̂``"
+   calcFq! :: Function
+
+  # derived params
+    "the reduced gravity constants for the fluid interface"
+        g′ :: T
+    "array containing x-gradient of PV due to eta in each fluid layer"
+        Qx :: Aphys3D
+    "array containing y-gradient of PV due to β, U, and eta in each fluid layer"
+        Qy :: Aphys3D
+    "rfft plan for FFTs"
+  rfftplan :: Trfft
+end
```

Comment:
It seemed to me we could have `NLayerParams{N, ...}` to avoid copy-paste for `SingleLayerParams` (or even better, just add `N` to `Params`...) This is totally fine with me now if it works, we can clean up later.